### PR TITLE
Continuous Integration with automated tests and coverage

### PR DESCRIPTION
**Pull request by niklasp**
For continuing at plastex development it is useful to have automated tests running. I added Travis CI as a continuous integration solution, that will automatically test pushes **and pull requests**. 

Also added automatic coverage report, which will help us in writing tests, by seeing what needs to be covered in the tests.

You can see the working versions for my fork here:
https://travis-ci.org/niklasp/plastex
https://coveralls.io/github/niklasp/plastex

The repo admins need to register at their github account at Travis and Coveralls, and [enable plastex repo in Travis](https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI) [as well as in coveralls](https://coveralls.io/repos/new). 

Finally I added badges to the README showing build and coverage status ([see here](https://github.com/niklasp/plastex/tree/python3#status)) Admins need to update the urls to the this repo (should just be replacing `niklasp` by `tiarno`.

 :sparkles: :sparkles: :sparkles:

### CODE REVIEW

File: unittests/Packages/Longtable.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def Soup(source):
 except ImportError:
     from BeautifulSoup import BeautifulSoup as Soup
 
-
+@unittest.skip("skip class for now as it uses the binary")
```
Comment (niklasp) [inline unittests/Packages/Longtable.py]: This needs more attention. My push request is just a starting point. Did not look into test code yet.

**Comment by niklasp** [comment]
👍 
Could you still register the repo with coveralls and travisCI, that would be needed for automated tests and coverage reports. I'll then update the links.

**Comment by kesmit13** [comment]
@tiarno will have to do that since he owns the repository.

**Comment by tiarno** [comment]
thanks everyone, I'm working on registering with travisCI and coveralls now. Seems like the network is slow but if it doesn't make it today I'll get back to it tomorrow.

**Comment by tiarno** [comment]
well that's a start at least. registered with travisCI and coveralls but failing pytest and the coveralls badge markup must have gotten munged in the paste. The .travis.yml specifies python2.7 since the master branch is still at python2, not python3.

**Comment by kesmit13** [comment]
It may just be that it interprets "README" as plain text, and the URL requires markdown.

**Comment by niklasp** [comment]
yeah, i just wrote it for the python3 branch, never worked with 2, quite new to the game i am.

**Comment by tiarno** [comment]
That was right, the README wasn't parsed for markdown, now it is. I remember a question from a while back about making the python3 version master. @niklasp is that what your tests are against? @kesmit13 what do you think about changing master? feel free to contact off line.

**Comment by kesmit13** [comment]
If python3 is the version everyone is making enhancements to and testing, I can't think of a reason not to make that the version in master.

**Comment by niklasp** [comment]
yes, i am trying to build a robust test setup, before I would start developing any new features / fixes. What is the status of python3 testing? I remember that you, @PatrickMassot, said something about it but cannot find it anymore. I am trying to get all existing tests running right now but am failing for some tests. My aim also is to not run with the plastex script on a process but rather find a solid way to running it as a library and getting all dependencies and packages right. Happy about help and/or coordination.

**Comment by tiarno** [comment]
okay, I created a new branch from master called python2 and made python3 the default. I would have changed the name from python3 to master but my git-fu is not strong enough. @niklasp, I will put together a little example of using plastex as a lib. I never use the command line script.

**Comment by tiarno** [comment]
@niklasp, this is untested, and you should fill out your own environment variable paths. 
This should give you an idea of using it as a lib. 
I use plastex to create DocBook, you probably want to change the renderer as well.
This is taken from a more complicated set up where there are classes for configuration and compiling.

    from plasTeX.Config import config as PLconfig
    from plasTeX.Renderers.DocBook import Renderer
    from plasTeX.TeX import TeX
    import plasTeX
   
      def set_defaults():
           cfg = PLconfig
           cfg['general']['renderer'] = 'DocBook'
           cfg['general']['copy-theme-extras'] = False
           cfg['general']['theme'] = 'default'
   
           cfg['images']['imager'] = 'dvipng'
           cfg['images']['save-file'] = True
           cfg['images']['transparent'] = True
           cfg['images']['enabled'] = True
   
           cfg['files']['bad-chars-sub'] = '_'
           cfg['files']['output-encoding'] = 'UTF-8'
   
           return cfg
  
    def main(tex_name):
      with open('plastex.ini'), 'w') as f:
          f.write('%r' % set_defaults())
      ini_file = 'plastex.ini'
      cfg = PLconfig
      cfg.read(self.ini_file)
      os.environ['TEXINPUTS'] =
      os.environ['PYTHONPATH'] =
      os.environ['DocBookTEMPLATES'] =
   
      document = plasTeX.TeXDocument(config=cfg)
      tex = TeX(document, file=tex_name)
      tex.parse()
      r = Renderer()
      r.render(document)

**Comment by PatrickMassot** [comment]
@tiarno since you mentioned, docbook, would you have time to handle #61?

**Comment by tiarno** [comment]
@PatrickMassot , yes. not immediately but it is on my list. thanks.

**Pull request merged**